aws/build.rake

```diff
@@ -166,45 +154,32 @@ if (rack_env?(:staging) && CDO.name == 'staging') || rack_env?(:development)
       RakeUtils.system 'rm', '-f', deploy_dir('rebuild-apps')
     end
   end
+else
+  APPS_COMMIT_TASK = build_task('apps-commit') {}
+end
 
-  #
-  # Define the CODE_STUDIO COMMIT task. If CODE_STUDIO_COMMIT_TASK produces new output, that output needs to be
-  #   committed because it's input for the DASHBOARD task.
-  #
-  CODE_STUDIO_COMMIT_TASK = build_task('code-studio-commit', [deploy_dir('rebuild'), CODE_STUDIO_TASK]) do
-    code_studio_changed = false
-    Dir.chdir(dashboard_dir('public/code-studio-package')) do
-      code_studio_git_status = `git status --porcelain .`
-      code_studio_changed = !code_studio_git_status.strip.empty?
-      HipChat.log "<b>code-studio</b> package changes:\n#{code_studio_git_status}" if code_studio_changed
-    end
+#
+# Define the CODE STUDIO BUILD task.
+#
+CODE_STUDIO_NODE_MODULES = Dir.glob(code_studio_dir('node_modules', '**/*'))
+CODE_STUDIO_BUILD_PRODUCTS = [code_studio_dir('npm-debug.log')] + Dir.glob(code_studio_dir('build', '**/*'))
+CODE_STUDIO_SOURCE_FILES = Dir.glob(code_studio_dir('**/*')) - CODE_STUDIO_NODE_MODULES - CODE_STUDIO_BUILD_PRODUCTS
+CODE_STUDIO_TASK = build_task('code-studio', CODE_STUDIO_SOURCE_FILES) do
+  packager = S3Packaging.new('code-studio', code_studio_dir, dashboard_dir('public/code-studio-package'))
 
-    if code_studio_changed
-      if RakeUtils.git_updates_available?
-        # NOTE: If we have local changes as a result of building CODE_STUDIO_TASK, but there are new
-        # commits pending in the repository, it is better to pull the repository first and commit
-        # these changes after we're caught up with the repository because, if we committed the changes
-        # before pulling we would need to manually handle a "merge commit" even though it's impossible
-        # for there to be file conflicts (because nobody changes the files CODE_STUDIO_TASK builds manually).
-        HipChat.log '<b>code-studio</b> package updated but git changes are pending; commmiting after next build.', color: 'yellow'
-      else
-        HipChat.log 'Committing updated <b>code-studio</b> package...', color: 'purple'
-        RakeUtils.system 'git', 'add', '--all', dashboard_dir('public/code-studio-package')
-        HipChat.log "<b>code-studio</b> staged changes:\n#{`git status --porcelain .`}"
-        message = "Automatically built.\n\n#{IO.read(deploy_dir('rebuild-code-studio'))}"
-        RakeUtils.system 'git', 'commit', '-m', Shellwords.escape(message)
-        RakeUtils.git_push
-        RakeUtils.system 'rm', '-f', deploy_dir('rebuild-code-studio')
-      end
-    else
-      HipChat.log '<b>code-studio</b> package unmodified, nothing to commit.'
-      RakeUtils.system 'rm', '-f', deploy_dir('rebuild-code-studio')
-    end
-  end
+  next if packager.update_from_s3
```

Comment:
What's the right way to break out of a do block like this in ruby? I wasn't sure, did some googling, and next seemed to work...